modules/rostests/apitests/ntdll/NtQueryInformationToken.c

```diff
@@ -0,0 +1,622 @@
+/*
+ * PROJECT:         ReactOS API tests
+ * LICENSE:         GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:         Tests for the NtQueryInformationToken API
+ * COPYRIGHT:       Copyright 2022 George Bișoc <george.bisoc@reactos.org>
+ */
+
+#include "precomp.h"
+
+static
+HANDLE
+OpenCurrentToken(VOID)
+{
+    BOOL Success;
+    HANDLE Token;
+
+    Success = OpenProcessToken(GetCurrentProcess(),
+                               TOKEN_READ | TOKEN_QUERY_SOURCE | TOKEN_DUPLICATE,
+                               &Token);
+    if (!Success)
+    {
+        skip("OpenProcessToken() has failed to get the process' token (error code: %lu)!\n", GetLastError());
+        return NULL;
+    }
+
+    return Token;
+}
+
+static
+VOID
+QueryTokenUserTests(
+    _In_ HANDLE Token)
+{
+    NTSTATUS Status;
+    PTOKEN_USER UserToken;
+    ULONG BufferLength;
+/*
+ * FIXME: Make this stuff compile only on MSVC or Clang
+ * because apparently GCC doesn't understand what "%wZ"
+ * conversion type is on trace().
+ */
+#if defined(_MSC_VER) || (__clang__)
+    UNICODE_STRING SidString;
+#endif
+
+    Status = NtQueryInformationToken(Token,
+                                     TokenUser,
+                                     NULL,
+                                     0,
+                                     &BufferLength);
+    ok_hex(Status, STATUS_BUFFER_TOO_SMALL);
+
+    UserToken = RtlAllocateHeap(RtlGetProcessHeap(), 0, BufferLength);
+    if (!UserToken)
+    {
+        skip("Failed to allocate heap for token user!\n");
```

Comment:
I guess a fail is more in place here,
and it would make sense to include the `BufferLength` in the logging, because this seems like it should not happen.
And a nit on the text: you are not allocating a heap, you are allocating from the heap.